### ISSUE TITLE
Interpreter Backend: Handling of mutable data in `tests/calls_on_ref_and_val_target/calls.fz`

### ISSUE THREAD

**Issue by fridis** [tokiwa-software/fuzion#1437]
Some of the tests fail resulting in output such as
```
case4a: FAILED: f#1 in oa type calls.case4.oa /= f#2 in oa type calls.case4.oa
case4a: FAILED: f#1 in oa type calls.case4.oa /= f#3 in oa type calls.case4.oa
case4a: FAILED: 0 /= 3
```
Need to check if this is due to #1436 or a bug in the interpreter backend.

**Comment by michaellilltokiwa** [comment]
fixed #2894

**Comment by michaellilltokiwa** [comment]
fixed #2894
